pipeline/tools/loader.py

```diff
@@ -49,10 +50,16 @@ def get_tool_path(loader, node):
 yaml.add_constructor("!get_tool_path", get_tool_path)
 yaml.add_constructor("!get_parent", get_parent)
 
-
-tools = {}
 for file in definitions.iterdir():
-    if file.name.endswith(".yaml"):
-        config = yaml.full_load(file.read_text())
+    if file.name.endswith(".yaml") and file.name not in tools:
+        try:
+            config = yaml.full_load(file.read_text())
+        except KeyError as error:  # load dependencies first
+            dependency = error.args[0]
+            dependency_file = definitions / (dependency + ".yaml")
+            dependency_config = yaml.full_load(dependency_file.read_text())
+            tools[dependency] = dependency_config
+            config = yaml.full_load(file.read_text())
+
```

Comment:
this seems fragile in the long term.  imagine we have two or three key dependencies in one yaml, this wouldn't catch it.  

what do you think about 

```python
def load_yaml(yamlfile):
    try:
        config = yaml.full_load(yamlfile.read_text())
        tool_name = str(file.name.replace(".yaml", ""))
        tools[tool_name] = config
    except KeyError as error:  # load dependencies first
        dependency = error.args[0]
        dependency_file = definitions / (dependency + ".yaml")
        load_yaml(dependency_file)
 
for file in definitions.iterdir():
    if file.name.endswith(".yaml") and file.name not in tools:
        load_yaml(file)
```

Reply:
modified to add extra load_yaml(file) after dependency exception, or else original file is never installed if missing dependency.